File: support-objects/src/main/java/com/yorosis/yoroapps/grid/vo/GridVO.java
package com.yorosis.yoroapps.grid.vo;

import java.util.List;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GridVO {
	private UUID gridId;
	private UUID pageId;
	private String gridName;
	private String moduleName;
	private int widthPercentage;
	private String filterable;
	private String gridUrl;
	private String defaultSortableColumn;
	private String showCheckBox;
	private String passParams;
	private Boolean userSpecificGridData;
	private String fieldValues;
	private String gridColumnNames;

	private List<GridColumnsVO> gridColumns;

	private List<GridFilterVO> permanentFilterColumns;
	private List<GridFilterVO> globalFilterColumns;

	private List<UUID> deletedColumnIDList;
	private List<UUID> deletedPermanentFilterIdList;
	private List<UUID> deletedGlobalFilterIdList;

	private int defaultNoOfRows;
	private String exportable;
	private String sortDirection;
}

File: yoroflow-ui/src/app/creation-module/grid-configuration/grid-vo.ts
// import { FieldConfig } from '../shared/vo/page-vo';

export class GridColumns {
    id: string;
    columnName: string;
    objectFieldName: string;
    displayName: string;
    sortable: boolean;
    widthPercentage: string;
    filterable: string;
    fieldType: string;
    columnSequenceNo: number;
    hiddenValue: string;
    defaultSortableColumn: boolean;
    passParams: boolean;
    asc: boolean;
    desc: boolean;
    dateTimeFormat: string;
}

export class GridFilterVO {
    filterId: string;
    filterName: string;
    operator: string;
    filterType: string;
    filterValue: string;
}


export class Grid {
    gridId: string;
    gridName: string;
    moduleName: string;
    widthPercentage: number;
    filterable: string;
    gridUrl: string;
    defaultSortableColumn: string;
    showCheckBox: string;
    passParams: string;
    sortDirection: string;
    gridColumns: GridColumns[] = [new GridColumns()];
    permanentFilterColumns: GridFilterVO[];
    globalFilterColumns: GridFilterVO[];
    deletedColumnIDList: string[] = [];
    deletedPermanentFilterIdList: string[] = [];
    deletedGlobalFilterIdList: string[] = [];
    defaultNoOfRows: number;
    exportable: string;
    userSpecificGridData: boolean;
    fieldValues: string;
    gridColumnNames: any[];
    // fieldConfig: FieldConfig;
}

export class GridColumnNames {
    label: string;
    name: string;
    dataType: string;
}
